resources/python/pipelines/pipelines.yml

```diff
@@ -50,11 +50,31 @@ jobs:
     displayName: Generate all explicit_requirements & Update source requirements
     workingDirectory: resources/python
 
+- job: install_source_dependencies_3_10
+  displayName: Install source dependencies Python 3.10
```

Comment:
Minor: could you move this block up above Py39 to keep the same DESC order please?
This should also simplify the changes of "dependsOn"

Reply:
I think doing so also will need to change the order between 3.7 and 3.9 we currently had.

I understand your POV. However, I think the blocks could be in the actual execution order to ease the debugging. Do you think it makes sense?

Reply:
It does not matter. It would just have helped the review process. I'll let you decide what you think is best.

Reply:
Let's keep the execution order for now until we simplify this a bit.